### PR TITLE
fix(NODE-3326): binding.gyp missing from files list in package.json

### PR DESCRIPTION
**Pull request by nbbeeken**
- Add test to ensure package contains exactly the files listed in an allow list

### CODE REVIEW

File: bindings/node/package.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
     "CHANGELOG.md",
     "lib",
     "src",
-    "build",
```
Comment (nbbeeken) [inline bindings/node/package.json:13]: @mbroadst @addaleax Hey, does this seem right to y'all? It's my understanding that the goal of prebuild is to either download or compile the package onsite, currently the build directory ships with a compiled version that matches the releaser's computer.

Comment (mbroadst) [reply bindings/node/package.json:13]: LGTM,  this must have been left in from early testing. Do you need to include the `binding.gyp` for that matter?

Comment (nbbeeken) [reply bindings/node/package.json:13]: If a prebuild isn't available then it falls back on attempting to compile the module and this file is needed to do so.

Comment (mbroadst) [reply bindings/node/package.json:13]: great point :facepalm:

File: bindings/node/test/release.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    fs.unlinkSync(packFile);
+  });
+
+  for (const requiredFile of REQUIRED_FILES) {
+    it(`should contain ${requiredFile}`, () => {
+      expect(tarFileList).to.includes(requiredFile);
+    });
+  }
+
+  it('should not have extraneous files', () => {
```
Comment (emadum) [inline bindings/node/test/release.test.js:53]: Looks like this test would make the above tests redundant; we could just call it something like 'should contain the correct files'.

**Review by emadum — APPROVED** [review]
Had one minor comment, but LGTM 👍

File: bindings/node/test/release.test.js (hunk 2 above)
Comment (nbbeeken) [reply bindings/node/test/release.test.js:53]: Updated the test if yall wanna take a look

**Review by dariakp — APPROVED** [review]

**Pull request merged**